src/structs.jl

```diff
@@ -396,9 +468,31 @@ function taxes(S::ScalarSector,H::ScalarConsumer)
     return P.taxes[H]
 end
 
+struct IndexedProduction{N} <: AbstractArray{ScalarProduction, N}
+    sector::IndexedSector
+    scalar_productions::AbstractArray{<:Union{ScalarProduction, Nothing},N}
+    index::Any
+    #IndexedProduction(sector::IndexedSector, scalar_productions::AbstractArray{<:UnionScalarProduction,N}, index) where N = new{N}(sector, scalar_productions, index)
```

Comment:
The scalars in the indexed productions are already added in ScalarProduction construction